src/wp-includes/class-wp-scripts.php

```diff
@@ -863,6 +863,18 @@ private function get_dependents( $handle ) {
 		return $dependents;
 	}
 
+	/**
+	 * Determines if a scripts defined loading strategy is valid.
+	 *
+	 * @param string $strategy The defined script strategy.
+	 * @return bool True if the strategy is of an allowed type, false otherwise.
+	 */
+	private function is_valid_strategy( $strategy ) {
+		$allowed_strategies = array( 'blocking', 'defer', 'async' );
```

Comment:
Could a filter be applied here so plugins could add experimental strategies like 'worker'? Could also be added later, but I'd love to ensure this is extensible.

Reply:
Personally, I'd be ok with leaving this as is for the initial release and add the additional filter based on developer feedback rather than adding it already. I believe if someone really had a use case where they were wanting to support additional strategies, they would be better off extending the whole class and replacing the global `WP_Scripts` object, rather than trying to filter Core's implementation.

Agree that a filter here _could_ be useful, but we would also need to consider the side effects if someone were to filter this list to remove some/all of our supported strategies as a defensive move. Perhaps only make the filter additive, like this pseudocode example?

```php
$additional_strategies = apply_filter( 'wp_add_script_loading_strategy', array() )

$allowed_strategies  = array_merge( array( 'blocking', 'defer', 'async' ), $additional_strategies );
```

Reply:
I share @joemcgill's sentiment here, but from a slightly different perspective of what does it mean for the rest of the process? We do a lot of _deciding_ on what the most eligible/applicable strategy for a script is based on the known strategies of defer, async or blocking. What does it means for the dependency tree when a developer introduces a custom/other strategy? Does it mean we need to abort any logical checks and simply output the custom strategy? I have the same question for inline script handling.

Reply:
I agree with @joemcgill here. There were many instances in the past were people create plugin to stop a feature, classic example is Classic them plugins for Gutenberg. Even though this is not same case, there will some people who would want to seize this opportunity and build custom plugins to do so. Core strategies should possibly not be allowed to be remove at all, as they can opt out by not using any strategy. Again as @10upsimon rightly suggested there are lot more to consider in terms of expected behaviour before we decide to add this filter.